cat/opencl.cat

```diff
@@ -0,0 +1,128 @@
+OpenCL
+(* OpenCL Memory Model *)
+include "basic.cat"
+
+(*
+* This model is based on:
+* https://multicore.doc.ic.ac.uk/overhauling/opencl_base.cat
+* https://multicore.doc.ic.ac.uk/overhauling/opencl_scopedsc.cat
+*)
+
+// Base relations:
+// int: same work-item
+// swg: same work-group
+// sdv: same device
+// syncbar: same barrier id
+
+// Tags:
+// WI: work-item scope
+// WG: work-group scope
+// DV: device scope
+// ALL: all-svm-devices scope
+// GLOBAL: global memory
+// LOCAL: local memory
+// EF: entry fence of barrier
+// XF: exit fence of barrier
+
+// dynamic_tag relates events to itself that access an address whose init event is marked X or Fence tagged with X
+let dynamic_tag(X) = [range([IW & X]; loc)] | [X & F]
+
+let mo = dynamic_tag(~NAL); co; dynamic_tag(~NAL)
+let sb = po
+let rb = (rf^-1; mo) \ id
+let unv = _ * _
+let wi = int
+
+(* Inclusive scopes *)
+let incl = (swg & (WG * WG)) | (sdv & (DV * DV)) | (ALL * ALL)
+
+(*******************)
+(* Synchronisation *)
+(*******************)
+
+let Acq = (ACQ | SC | ACQ_REL) & (R | F)
+let Rel = (REL | SC | ACQ_REL) & (W | F)
```

Comment:
A like the names `Acq` and `Rel`. Can you please rename `cacq` and etc in the c11 variants to make them patch this?

Reply:
Renamed